src/Microsoft.AspNet.Mvc.Core/ViewComponents/ViewComponent.cs

```diff
@@ -99,13 +102,60 @@ public ModelStateDictionary ModelState
         /// Gets or sets the <see cref="ViewContext"/>.
         /// </summary>
         [Activate]
-        public ViewContext ViewContext { get; set; }
+        public ViewContext ViewContext
+        {
+            get
+            {
+                if (_viewContext == null)
+                {
+                    // This should run only for the ViewComponent unit test scenarios
+                    _viewContext = new ViewContext();
+                    _viewContext.ViewData = new ViewDataDictionary(
+                        new EmptyModelMetadataProvider(),
+                        _viewContext.ModelState);
```

Comment:
ViewContext is shared everywhere..and I thought we might change that default constructor again when we find new requirements when considering unit testability of like TagHelper, HtmlHelper, etc..So, I think a minimum change should be better at this time..  
@dougbu What do you think??